backend/utils.py

```diff
@@ -18,10 +20,55 @@
 from backend.config import CONFIG
 
 
-def powerset(iterable):
-    """powerset([1,2,3]) --> () (1,) (2,) (3,) (1,2) (1,3) (2,3) (1,2,3)"""
-    s = list(iterable)
-    return chain.from_iterable(combinations(s, r) for r in range(len(s)+1))
+def debounce(wait):
+    """ Decorator that will postpone a function's
+        execution until after `wait` seconds
+        have elapsed since the last time it was invoked.
+        from https://chat.openai.com/share/7ed01b5b-97bd-436a-b23b-542f068a4302"""
+    def decorator(func):
+        def debounced(*args, **kwargs):
+            def call_it():
+                func(*args, **kwargs)
+            if hasattr(debounced, '_timer'):
+                debounced._timer.cancel()
+            debounced._timer = threading.Timer(wait, call_it)
```

Comment:
This is interesting. I have no idea how this works. It's weird to see the name of the function referenced inside of itself.
Otherwise looks familiar to how I've done decorators before.

I don't see this being used yet, so I'm guessing it might not be fully functional yet?